docs/concepts/filesystem-semantics.md

```diff
@@ -0,0 +1,102 @@
+# Mountpoint for Amazon S3 File System Behavior
+
+The Scality S3 CSI Driver uses Mountpoint for Amazon S3 to present S3-compatible buckets as filesystems.
+All file system semantics are inherited from Mountpoint for Amazon S3, and are compatible with Scality S3 and any S3-compatible storage.
+This documentation describes only the behavior relevant to S3 Standard buckets.
+Features specific to AWS services other than S3, such as S3 Express One Zone, Glacier, or Archive storage classes, are not supported or relevant.
+
+## Behavior Tenets
+
+1. Mountpoint does not support file behaviors that cannot be implemented efficiently against S3's object APIs. It does not emulate operations like `rename` that would require many API calls to S3 to perform.
+2. Mountpoint presents a common view of S3 object data through both file and object APIs.
+   It does not emulate POSIX file features that have no close analog in S3's object APIs, such as mutable ownership and permissions.
+3. When these tenets conflict with POSIX requirements, Mountpoint fails early and explicitly.
+   It will cause applications to fail with IO errors rather than silently accept operations that will never succeed, such as extended attributes.
+
+## Reading and Writing Files
+
+- Supports opening and reading existing objects, optimized for large sequential reads.
+- Supports random reads and seeking within existing objects.
+- Supports creating new objects by writing new files.
+- With `--allow-overwrite`, supports replacing existing objects (must use `O_TRUNC` to truncate).
+- Writes must always start from the beginning and be sequential.
+- Uploads are asynchronous and optimized for throughput; use `fsync` to guarantee upload before closing.
+- Cannot continue writing after `fsync`.
+- New or overwritten objects are visible to other S3 clients only after closing or `fsync`.
+- By default, deleting objects (`rm`) is not allowed; enable with `--allow-delete`.
+- Delete operations immediately remove the object from S3.
+- Cannot delete a file while it is being written.
+- Renaming files is not supported.
+
+## Directories
+
+- S3 is flat; Mountpoint infers directories from `/` in object keys.
+- Not all S3 object keys correspond to valid file names; some objects may not be accessible.
+- If a directory and file share a name, only the directory is accessible.
+- Creating directories (`mkdir`) is local and not persisted to S3 until a file is written inside.
+- Cannot remove or rename existing directories; can remove new local directories if empty.
+- No support for hard or symbolic links.
```

Comment:
And I suppose as well any other special unix file type such as sockets, FIFO named pipes, block or character special files,